### PR TITLE
make pulsar executor pool size configurable

### PR DESCRIPTION
**Pull request by hangc0276**
#### Motivation
The pulsar executor pool size number is hard code to `20` when pulsar service start, it should be configurable in broker.conf.
```
private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(20,
            new DefaultThreadFactory("pulsar"));
```

#### Changes
make the executor pool size configurable in broker.conf

### CODE REVIEW

File: conf/broker.conf [hunk 1]
```diff
@@ -61,6 +61,12 @@ numIOThreads=
 # Number of threads to use for HTTP requests processing. Default is set to 2 * Runtime.getRuntime().availableProcessors()
 numHttpServerThreads=
 
+# Number of thread pool size to use for pulsar broker service.
+# The executor in thread pool will do basic broker operation like load/unload bundle, update managedLedgerConfig,
+# update topic/subscription/replicator message dispatch rate, do leader election etc.
+# Default is 20
+numExecutorThreadPoolSize=
```
Comment (merlimat) [inline conf/broker.conf:68]: I would be better to default to something like `Runtime.getRuntime().availableProcessors()`

**Comment by merlimat** [comment]
As a followup to these changes in thread pool sizes, it would be nice to reduce the size of these pools when running the unit tests

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 2]
```diff
@@ -170,8 +170,8 @@
     private LocalZooKeeperConnectionService localZooKeeperConnectionProvider;
     private Compactor compactor;
 
-    private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(20,
-            new DefaultThreadFactory("pulsar"));
```
Comment (jiazhai) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:174]: @hangc0276  would you please help changed this value as Matteo suggested? `Runtime.getRuntime().availableProcessors()`

Comment (hangc0276) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:174]: Thanks for your review, i will update the pr.

File: conf/broker.conf (hunk 1 above)
Comment (hangc0276) [reply conf/broker.conf:68]: ok, it's better to change it to Runtime.getRuntime().availableProcessors(). it will update the pr.

**Comment by hangc0276** [comment]
> As a followup to these changes in thread pool sizes, it would be nice to reduce the size of these pools when running the unit tests

Yes, for the unit tests framework, i will reduce the pool size.

**Comment by hangc0276** [comment]
> > As a followup to these changes in thread pool sizes, it would be nice to reduce the size of these pools when running the unit tests
> 
> Yes, for the unit tests framework, i will reduce the pool size.

@merlimat @jiazhai  I have changed the default executor pool size to Runtime.getRuntime().availableProcessors() and update unit test executor pool size to 5, please take a look, thanks.

**Pull request merged**